smart-contracts/ethereum/contracts/UserProfile.sol

```diff
@@ -1,65 +1,126 @@
-pragma solidity ^0.4.15;
-
-/**
- * The userProfile contract:
- * - Verify users
- * - access list of the installed apps
- * - data refs
- */
+pragma solidity >=0.4.15;
+
+import "./MarketPlace.sol";
+
 contract UserProfile {
-  struct appDetails {
+
+  struct App {
     uint id; // redundant but usefull to include
-    uint version;
+    bytes32 version;
     bool enabled;
     uint permissions; // bitmap for effeciency ;)
   }
 
   struct Key {
-    string public_key; // it can be any blockchain address 
-    string private_key;
+    string publicKey; // it can be any blockchain address 
+    string privateKey;
```

Comment:
Think we could start basic documentation (simple docs/*.md files) around what these fields are for? Key struct in particular is a bit unclear